doc/source/whatsnew/v1.0.0.rst

```diff
@@ -421,6 +421,7 @@ Reshaping
 - :func:`qcut` and :func:`cut` now handle boolean input (:issue:`20303`)
 - Fix to ensure all int dtypes can be used in :func:`merge_asof` when using a tolerance value. Previously every non-int64 type would raise an erroneous ``MergeError`` (:issue:`28870`).
 - Better error message in :func:`get_dummies` when `columns` isn't a list-like value (:issue:`28383`)
+- Bug in :meth:`Index.join` that caused infinite recursion error for mismatched multiindex name orders (:issue:`25760`, :issue:`28956`)
```

Comment:
Can you make this more user-facing? I somehow doubt the test case in this PR is something users would do in actual code, so this seems to be hinting at addressing some other use case(s)

Reply:
putting a use case into the docs or expanding the tests?